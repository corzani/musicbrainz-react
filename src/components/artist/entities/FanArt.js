import {optional as opt} from "../../../common";
import React from "react";
import Grid from "@material-ui/core/Grid/Grid";
import Card from "@material-ui/core/Card/Card";
import CardMedia from "@material-ui/core/CardMedia/CardMedia";
import PropTypes from "prop-types";
import {withStyles} from "@material-ui/core";
import {styles} from '../styles';

const fanArtComponent = ({fanArt, classes}) => {
    const {thumbnails = []} = fanArt || {};
    const imageURLs = thumbnails.map(thumbnail => thumbnail.url);
    const cover = path => opt(path, '/no-cover.png');

    return (
        <React.Fragment>
            {(imageURLs.length === 0) ? <p>No Fan Art found.</p> : null}
            {imageURLs.map((url, index) => (
                <Grid item key={`album-${index}`} sm={6} md={6} lg={6}>
                    <Card className={classes.card}>
                        <CardMedia
                            className={classes.cardMedia}
                            image={cover(url)}
                        />
                    </Card>
                </Grid>
            ))}
        </React.Fragment>)
};

fanArtComponent.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(fanArtComponent);

