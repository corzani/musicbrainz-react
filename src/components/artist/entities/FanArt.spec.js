import {render} from 'enzyme/build';
import FanArt from './FanArt';
import {pinkFloydArtist} from '../../../test/pinkfloydResponse';
import toJson from 'enzyme-to-json';
import React from 'react';

describe('FanArt component', () => {
    it('should show information when "fanArt" property is given', () => {
        const lookup = render(<FanArt fanArt={pinkFloydArtist.fanArt}/>);
        expect(toJson(lookup)).toMatchSnapshot();
    });

    it('should show default information when "fanArt" property is not given', () => {
        const lookup = render(<FanArt fanArt={{}}/>);
        expect(lookup.text()).toEqual('No Fan Art found.');
    });

    it('should show default image when no cover art is present', () => {
        const fanArtWithNoCover = {
            "thumbnails": [{}]
        };

        const lookup = render(<FanArt fanArt={fanArtWithNoCover}/>);
        expect(toJson(lookup)).toMatchSnapshot();
    });
});