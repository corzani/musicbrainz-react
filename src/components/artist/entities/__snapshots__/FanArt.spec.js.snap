// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FanArt component should show default image when no cover art is present 1`] = `
<div
  class="MuiGrid-item-13 MuiGrid-grid-sm-6-60 MuiGrid-grid-md-6-74 MuiGrid-grid-lg-6-88"
>
  <div
    class="MuiPaper-root-110 MuiPaper-elevation1-113 MuiPaper-rounded-111 MuiCard-root-109 fanArtComponent-card-9"
  >
    <div
      class="MuiCardMedia-root-137 fanArtComponent-cardMedia-10"
      style="background-image:url(\\"/no-cover.png\\")"
    />
  </div>
</div>
`;

exports[`FanArt component should show information when "fanArt" property is given 1`] = `
Array [
  <div
    class="MuiGrid-item-13 MuiGrid-grid-sm-6-60 MuiGrid-grid-md-6-74 MuiGrid-grid-lg-6-88"
  >
    <div
      class="MuiPaper-root-110 MuiPaper-elevation1-113 MuiPaper-rounded-111 MuiCard-root-109 fanArtComponent-card-9"
    >
      <div
        class="MuiCardMedia-root-137 fanArtComponent-cardMedia-10"
        style="background-image:url(\\"https://assets.fanart.tv/fanart/music/83d91898-7763-47d7-b03b-b92132375c47/artistthumb/pink-floyd-4fad917437146.jpg\\")"
      />
    </div>
  </div>,
  <div
    class="MuiGrid-item-13 MuiGrid-grid-sm-6-60 MuiGrid-grid-md-6-74 MuiGrid-grid-lg-6-88"
  >
    <div
      class="MuiPaper-root-110 MuiPaper-elevation1-113 MuiPaper-rounded-111 MuiCard-root-109 fanArtComponent-card-9"
    >
      <div
        class="MuiCardMedia-root-137 fanArtComponent-cardMedia-10"
        style="background-image:url(\\"https://assets.fanart.tv/fanart/music/83d91898-7763-47d7-b03b-b92132375c47/artistthumb/pink-floyd-4fb4eca9370e6.jpg\\")"
      />
    </div>
  </div>,
  <div
    class="MuiGrid-item-13 MuiGrid-grid-sm-6-60 MuiGrid-grid-md-6-74 MuiGrid-grid-lg-6-88"
  >
    <div
      class="MuiPaper-root-110 MuiPaper-elevation1-113 MuiPaper-rounded-111 MuiCard-root-109 fanArtComponent-card-9"
    >
      <div
        class="MuiCardMedia-root-137 fanArtComponent-cardMedia-10"
        style="background-image:url(\\"https://assets.fanart.tv/fanart/music/83d91898-7763-47d7-b03b-b92132375c47/artistthumb/pink-floyd-4fad9106aa7bf.jpg\\")"
      />
    </div>
  </div>,
]
`;
